src/librustc/session/config.rs

```diff
@@ -1145,6 +1145,20 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
 
     let debugging_opts = build_debugging_options(matches, error_format);
 
+    // Incremental compilation only works reliably when translation is done via
+    // MIR, so let's enable -Z orbit if necessary (see #34973).
+    let debugging_opts = if debugging_opts.incremental.is_some() &&
+                            !debugging_opts.orbit {
+        early_warn(error_format, "Automatically enabling `-Z orbit` because \
+                                  `-Z incremental` was specified");
+        DebuggingOptions {
+            orbit: true,
+            .. debugging_opts
+        }
```

Comment:
> This is a bit too much, could mutation be used instead?

You think? I actually thought that it was more readable that way. The alternative would be to make `debugging_opts` mutable, but it's within a *huge* function.

Or would you prefer this:
```rust
    let debugging_opts = if debugging_opts.incremental.is_some() &&
                            !debugging_opts.orbit {
        early_warn(error_format, "Automatically enabling `-Z orbit` because \
                                  `-Z incremental` was specified");
        let mut debugging_opts = debugging_opts;
        debugging_opts.orbit = true;
        debugging_opts
    }
```
I really don't care that much one way or the other, just tell me which version you want :)